src/bindings/c/src/CMakeLists.txt

```diff
@@ -5,7 +5,7 @@
 set(TARGET_NAME openvino_c)
 
 # Suppress warnings due to catch macro with legacy exception types
-ov_deprecated_no_errors()
+add_definitions(-DIN_OV_COMPONENT)
 
```

Comment:
Please add this back, since you have removed OPENVINO_SUPPRESS_DEPRECATED_START/END in ie_c_api.cpp